src/helpers/messaging.rs

```diff
@@ -121,6 +180,16 @@ impl Mesh<'_, '_> {
             )?;
         }
 
+        if let TotalRecords::Specified(count) = self.total_records {
```

Comment:
maybe add a method to `TotalRecords` to validate this as it is used in more than one place?